### PR TITLE
Feat/better offer volumes

### PR DESCRIPTION
**Pull request by adhusson**
tldr:
* `offer.gives` goes up to 160 bits
* marketOrder's `fillVolume` as well
* volume-based marketOrder & offer creation/updated are limited to 128 `[maker|taker][Wants|Gives]` (nontrivial & no time to allow the full price range with the API, which I'd say is fine since it exists mostly for convenience)

For `ratioFromVolumes`, use Remco Bloemen's [muldiv](https://xn--2-umb.com/21/muldiv/). For `inboundFromOutbound` and similar functions, use a new `mulDivPow2` that should be carefully reviewed.

**Why `mulDivPow2`**: `mulDiv(a,b,d)` (and `mulDivRoundUp`) return `a*b/d`. It throws if the result does not fit in 256 bits or if `d=0`. This is useful to compute a ratio from `gives` and `wants`. 

But to compute a `wants` from `gives` and `tick`, you start by converting a tick to a ratio represented as a `mantissa,exponent` pair, and `wants = gives*mantissa / 2^exponent`. The exponent may be > 255, so we can't use `mulDiv`. `mulDivPow2(a,b,e)` returns `a*b/(2<<e)`. It throws if the result does not fit in 256 bits. It also takes an extra `roundUp` argument.

In reviewing `mulDivPow2` you can assume that the first assembly block correctly returns the full `a*b` product, split into two words `prod1` and `prod0`.

### CODE REVIEW

**Comment by adhusson** [comment]
Closing in favor of #597
